packages/block-library/src/template-part/index.php

```diff
@@ -56,12 +65,36 @@ function render_block_core_template_part( $attributes ) {
 		}
 	}
 
-	if ( is_null( $content ) ) {
-		return 'Template Part Not Found';
+	if ( ! $template_part_id || is_null( $content ) ) {
+		return __( 'Template Part not found.' );
+	}
+
+	if ( isset( $seen_ids[ $template_part_id ] ) ) {
+		if ( ! is_admin() ) {
+			trigger_error(
+				sprintf(
+					// translators: %s are the block attributes.
+					__( 'Could not render Template Part block with the attributes <code>%s</code>: blocks cannot be rendered inside themselves.' ),
+					wp_json_encode( $attributes )
```

Comment:
It's hard to tell how what to show here. `$template_part_id` might be better but it can be different things.

Reply:
I'm not super worried about this. Whether we show attributes or an ID, the most important part of the UX will be for users to visually see which template part failed from the context, and when they go into the editor they will see the error state in the `Edit` element. I think either datum is fine.

Reply:
I think it reads good:
<img width="1210" alt="Screen Shot 2021-03-03 at 11 54 09" src="https://user-images.githubusercontent.com/699132/109804380-76df2580-7c22-11eb-8be0-e26f12fef6ca.png">